sys/pylon/gstpylonsrc.c

```diff
@@ -991,7 +991,9 @@ gst_pylonsrc_get_caps (GstBaseSrc * bsrc, GstCaps * filter)
     GST_DEBUG_OBJECT (src, "Could not send caps - no camera connected.");
     return gst_pad_get_pad_template_caps (GST_BASE_SRC_PAD (bsrc));
   } else {
-    return src->caps;
+    GstCaps* result = gst_caps_copy(src->caps);
```

Comment:
Good catch, in all the plugins I've written, I return a copy like your change does. I also notice that filter is ignored in this case, which I've used in my plugins, not entirely sure whether it's necessary. That can be a future commit.